src/SubjectFilter.php

```diff
@@ -365,19 +365,23 @@ protected function applySpec(Spec $spec, object $subject): bool
      *
      *
      */
-    protected function failed(Spec $spec): Failure
+    protected function failed(Spec $spec)
```

Comment:
This is one place I dislike.

Reply:
Is return void sufficient? The caller does not use the return value.

Reply:
Earlier it was returning the Failure. As it currently changed this now can return Failure or FailureCollection. ie why return is removed. I will look once again if this returned value has some value inside the caller.

Reply:
This method is to add a failure. It seems no return value is needed.